astro/src/content/docs/extend/code/lambdas/mfa-requirement.mdx

```diff
@@ -0,0 +1,315 @@
+     * see https://fusionauth.io/docs/reference/data-types#instants for instant documentation
+     * @type {Map<UUID, Instant>}
+     */
+    applications;
+
+    /**
+     * The first instant the trust was established for the user's tenant.
+     * see https://fusionauth.io/docs/reference/data-types#instants for instant documentation
+     * @type {Instant}
+     */
+    tenant;
+}
+
+class Trust {
+    /**
+     * The application that the MFA trust the user already has, was first associated with
+     * @type {string}
+     */
+    applicationId;
+
+    /**
+     * Additional attributes about the trust.
+     * @type {Map<String, Object>}
+     */
+    attributes;
+
+    /**
+     * When the MFA trust expires, see https://fusionauth.io/docs/reference/data-types#instants
+     * @type {Instant}
+     */
+    expirationInstant;
+
+    /**
+     * The twoFactorTrustId
+     * @type {String}
+     */
+    id;
+
+    /**
+     * When the MFA trust was created, see https://fusionauth.io/docs/reference/data-types#instants
+     * @type {Instant}
+     */
+    insertInstant;
+
+    /**
+     * Details about each application that trust was established for.
+     * @type {StartInstant}
+     */
+    startInstants;
+
+    /**
+     * Additional attributes/state trust.
+     * @type {Map<String, Object>}
+     */
+    state;
+
+    /**
+     * The user's tenant Id.
+     * @type {string}
+     */
+    tenantId;
+
+    /**
+     * The user's Id.
+     * @type {string}
+     */
+    userId;
+}
+```
+
+See [Application API](/docs/apis/applications) documentation for the `context.application` object details.
+
+See [Advanced Threat Detection](/docs/operate/secure/advanced-threat-detection) for more information on this licensed feature.
+
+## Assigning The Lambda
+
+Once a lambda is created, you must assign it to a Tenant. Using the FusionAuth admin UI, find the <InlineField>MFA requirement lambda</InlineField> field found on the <Breadcrumb>Multi-Factor</Breadcrumb> tab in the Tenant configuration.
+
+Or if you are using the Tenant API, assign the lambda Id to the `tenant.lambdaConfiguration.multiFactorRequirementId` field.
+
+A lambda can also be configured at the application level and, if an application is provided on the request (see action above), the application's lambda will take precedence over the tenant lambda. To assign an application specific lambda, using the FusionAuth admin UI, find the <InlineField>MFA requirement lambda</InlineField> field found on the <Breadcrumb>Multi-Factor</Breadcrumb> tab in the Application configuration.
+
+Or if you are using the Application API, assign the lambda Id to the `application.lambdaConfiguration.multiFactorRequirementId` field.
+
+## Example Lambdas
+
+Here is an example lambda that will require MFA if the user's email address contains `gilfoyle`. Otherwise, FusionAuth's out of the box decision will be used.
```

Comment:
```suggestion
The following example requires MFA if the user's email address contains the string `gilfoyle`, and otherwise uses the default MFA decision:
```